lib/ecto_job/worker.ex

```diff
@@ -22,7 +23,12 @@ defmodule EctoJob.Worker do
   def start_link(repo, job = %queue{}, now) do
     Task.start_link(fn ->
       with {:ok, job} <- JobQueue.update_job_in_progress(repo, job, now) do
-        queue.perform(JobQueue.initial_multi(job), job.params)
+        case queue.perform(JobQueue.initial_multi(job), job.params) do
+          {:ok, res} -> res
+          {:error, reason} -> JobQueue.update_error(repo, job, reason)
+          {:error, _ , message, changes_so_far} -> JobQueue.update_error(repo, extract_job(changes_so_far), message)
+          error -> raise "Unexpected return from job worker: #{error}"
+        end
```

Comment:
Unless I've missed it somewhere, it doesn't seem @henry-hz that there is any kind of try -> rescue that would catch exceptions coming out of `perform/2`. Having errors from the transaction persisted the way you have done would definitely be useful, but some users may be easily mislead since some errors would be persisted and others would be left to the logs. I would at least suggest that if you don't do this that you make it super explicit in the docs/readme that exceptions aren't saved.